File: python/6_kyu/dubstep.py
"""
SPDX-FileCopyrightText: 2025 Michael Persico <michael.a.persico@gmail.com>

SPDX-License-Identifier: MIT-0
"""

# https://www.codewars.com/kata/551dc350bf4e526099000ae5

def song_decoder(song):
    return " ".join([word for word in song.split("WUB") if len(word) != 0])
    
if __name__ == "__main__":
    print(song_decoder("AWUBBWUBC")) # A B C
    print(song_decoder("AWUBWUBWUBBWUBWUBWUBC")) # A B C
    print(song_decoder("WUBAWUBBWUBCWUB")) # A B C

File: java/6_kyu/Dubstep.java
/*
SPDX-FileCopyrightText: 2025 Michael Persico <michael.a.persico@gmail.com>

SPDX-License-Identifier: MIT-0
*/

// https://www.codewars.com/kata/551dc350bf4e526099000ae5

import java.util.*;
import java.util.stream.*;

public class Dubstep {

    public static String SongDecoder(String song) {
        return Arrays.asList(song.split("WUB")).stream().filter(i -> i.length() != 0).collect(Collectors.joining(" "));
    }
    
    public static void main(String[] args) {
        System.out.println(SongDecoder("WUBWUBABCWUB"));
        System.out.println(SongDecoder("RWUBWUBWUBLWUB"));
    }
}
